src/FTPObject.jl

```diff
@@ -111,6 +114,94 @@ end
 
 
 """
+    upload(
+        ftp::FTP,
+        local_file_paths::Vector{<:AbstractString},
+        ftp_dir<:AbstractString;
+        retry_callback::Function=(count, options) -> (count < 4, options),
+        retry_wait_seconds::Integer = 5
+    )
+
+Uploads the files specified in local_file_paths to the directory specifed by
+ftp_dir. The files will have the same names.
+
+By default, will try to deliver the files 4 times with a 5 second wait in between
+each failed attempt.
+
+You can specify a function for retry_callback to change behaviour. This function must
+take as parameters the number of attempts that have been made so far, and the current
+ftp connection options as a FTPClient.ConnContext type. It must return a boolean
+that is true if another delivery attempt can be made, and a TPClient.ConnContext
+type that is the connection options to use for all future files to be delivered. This
+allows backup ftp directories to be used for example.
+
+# Arguments
+`ftp::FTP`: The FTP to deliver to. See FTPClient.FTP for details.
+`file_paths::Vector{T}`: The file paths to the files we want to deliver.
+`ftp_dir`: The directory on the ftp server where we want to drop the files.
+`retry_callback::Function=(count, options) -> (count < 4, options)`: Function for retrying
+                                                                     when delivery fails.
+`retry_wait_seconds::Integer = 5`: How many seconds to wait in between retries.
+
+# Returns
+- `Array{Bool,1}`: Returns a vector of booleans with true for each successfully delivered
+                   file and false for any that failed to transfer.
+"""
+function upload(
+    ftp::FTP,
+    local_file_paths::Vector{<:AbstractString},
```

Comment:
This syntax is only available on Julia 0.6 and later. You should be able to do this if you write:
```julia
@compat function upload(
    ftp::FTP,
    local_file_paths::Vector{<:AbstractString},
    ...
```